virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -332,12 +335,38 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
     };
   }
 
+  private generateSessionEventsProvider(): Angular2PluginSessionEvents {
+    let loginSubject = new Subject<void>();
+    let logoutSubject = new Subject<void>();
+    let sessionExpireSubject = new Subject<void>();
```

Comment:
store these according to window id in a map

Reply:
@1000TurquoisePogs and I discussed, Each of the other events relate to the plugins at the plugin level, things like the window events & viewport events etc, but session events are for the session. It doesn't matter if you're in the Angular App or the Editor, a logout is still a logout so we could in theory have at max 3, but that would make it more difficult to control the events at lifecycle without tying them down to the window level. Do whatever you think is better @sakeerthy